src/components/Icon/Icon.scss

```diff
@@ -27,192 +27,394 @@
 }
 
 .isColored {
-  color: color('white');
+  color: var(--p-text-on-interactive, color('white'));
 }
 
 .colorWhite {
   @include color-icon('white');
   color: transparent;
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorBlack {
   @include color-icon('black');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorSkyLighter {
   @include color-icon('sky', 'lighter');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorSkyLight {
   @include color-icon('sky', 'light');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorSky {
   @include color-icon('sky');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorSkyDark {
   @include color-icon('sky', 'dark');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorInkLightest {
   @include color-icon('ink', 'lightest');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorInkLighter {
   @include color-icon('ink', 'lighter');
 
   &::before {
-    background-color: color('sky');
+    background-color: var(--p-surface, color('sky'));
+  }
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
   }
 }
 
 .colorInkLight {
   @include color-icon('ink', 'light');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorInk {
   @include color-icon('ink');
 
   &::before {
-    background-color: color('sky');
+    background-color: var(--p-surface, color('sky'));
+  }
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
   }
```

Comment:
Both `sky` and `ink` were tough to make calls on. `Ink` is a bit more straightforward, as it is always black, but `sky` is harder, since it is unclear to me how these are used, and if they are only used on buttons. If they're used on buttons, it is unnecessary to have white icon colors, since the button component will recolor any icons used to the appropriate color.

For now, I am going with all `sky` and `ink` icons aliasing to `--p-icon` in the new design language, though I could understand a case being made for making `sky` use white or inverse.

Reply:
We should flag this as part of the design review. Maybe make a note on the issue in polaris-ux?

Reply:
https://github.com/Shopify/polaris-ux/issues/348#issuecomment-593999923